### PR TITLE
feat: add moonbit http component

### PR DESCRIPTION
**Pull request by brooksmtownsend**
## Feature or Problem
This PR adds a moonbit http hello world component as per the instructions in https://www.moonbitlang.com/blog/component-model. Since the example still requires a wit-bindgen fork for the time being and I just started playing around with the language, I felt it fit here well!

It also updates the top level README to indicate the different projects we have in this repository 😄

### CODE REVIEW

File: secrets/secrets-kubernetes/go.sum [hunk 1]
```diff
@@ -43,7 +43,7 @@ github.com/kisielk/errcheck v1.5.0/go.mod h1:pFxgyoBC7bSaBwPgfKdkLd5X25qrDl4LWUI
 github.com/kisielk/gotool v1.0.0/go.mod h1:XhKaO+MFFWcvkIS/tQcRk01m1F5IRFswLeQ+oQHNcck=
 github.com/klauspost/compress v1.17.9 h1:6KIumPrER1LHsvBVuDa0r5xaG0Es51mhhB9BQB2qeMA=
 github.com/klauspost/compress v1.17.9/go.mod h1:Di0epgTjJY877eYKx5yC51cX2A2Vl2ibi7bDH9ttBbw=
-github.com/kr/pretty v0.2.1/go.mod h1:ipq/a2n7PKx3OHsz4KJII5eveXtPO4qwEXGdVfWzfnI=
+github.com/kr/pretty v0.2.0/go.mod h1:ipq/a2n7PKx3OHsz4KJII5eveXtPO4qwEXGdVfWzfnI=
```
Comment (joonas) [inline secrets/secrets-kubernetes/go.sum:46]: This seems like this might've snuck in inadvertently?

Comment (brooksmtownsend) [reply secrets/secrets-kubernetes/go.sum:46]: Dangit this keeps trying to update

Comment (brooksmtownsend) [reply secrets/secrets-kubernetes/go.sum]: reverted

Comment (joonas) [reply secrets/secrets-kubernetes/go.sum]: it could be the Go LSP getting triggered if you have the entire repo open

**Comment by lxfontes** [comment]
❤️  for the repo readme

**Review by joonas — APPROVED** [review]

**Pull request merged**